### PR TITLE
INS-3122: replace some ObjectReferences with RecordReferences

### PR DESCRIPTION
**Pull request by bigbes**
-- Record References are references with zeroed base part (and ends with .record in textual repr)
-- Object References (Local) have left and right part duplicated (and starts with 1 in textual repr)

* Replace Code/Reason/Request references with RecordReferences (all other references should be ObjectReferences)
* Add checks for ObjectReferences/RecordReferences in incoming messages
* Actualize unit+func tests

### CODE REVIEW

File: insolar/payload/payload.go [hunk 1]
```diff
@@ -139,6 +141,14 @@ func (h *MessageHash) IsZero() bool {
 	return true
 }
 
+func (m RequestInfo) RequestAsObjectReference() *insolar.Reference {
```
Comment (andreyromancev) [inline insolar/payload/payload.go]: Can we move these to logicrunner helpers? Doesn't have to be a method.

Comment (bigbes) [reply insolar/payload/payload.go]: Both of these functions? `RequestAsObjectReference`+`RequestReference`?

Comment (andreyromancev) [reply insolar/payload/payload.go]: Yep. Let's not pollute the record package.

Comment (bigbes) [reply insolar/payload/payload.go]: kk, pushed

File: insolar/payload/payload.go [hunk 2]
```diff
@@ -139,6 +141,10 @@ func (h *MessageHash) IsZero() bool {
 	return true
 }
 
+func (m RequestInfo) RequestReference() *insolar.Reference {
```
Comment (andreyromancev) [inline insolar/payload/payload.go]: Same here, let's move this somewhere else.

Comment (bigbes) [reply insolar/payload/payload.go]: Done; I've rewritten code, but forgot to remove old wrapper.

**Pull request merged**